Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp

```diff
@@ -212,68 +210,78 @@ static String dmabufRendererWithSupportedBuffers()
 
 #if USE(LIBDRM)
 
-// Cherry-pick function 'drmGetFormatName' from 'https://gitlab.freedesktop.org/mesa/drm/-/blob/main/xf86drm.c'.
-// Function is only available since version '2.4.113'. Debian 11 ships '2.4.104'.
-// FIXME: Remove when Debian 11 support ends.
-static char* webkitDrmGetFormatName(uint32_t format)
+// Base on function 'drmGetFormatName' from 'https://gitlab.freedesktop.org/mesa/drm/-/blob/main/xf86drm.c'.
+static String webkitDrmGetFormatName(uint32_t format)
 {
-    char* str;
-    char code[5];
-    const char* be;
-    size_t strSize, i;
-
-    // Is format big endian?
-    be = (format & (1U<<31)) ? "_BE" : "";
+    bool isBigEndian = (format & (1U << 31));
     format &= ~(1U<<31);
 
     // If format is DRM_FORMAT_INVALID.
     if (!format)
-        return strdup("INVALID");
+        return "INVALID"_s;
 
-    code[0] = (char) ((format >> 0) & 0xFF);
-    code[1] = (char) ((format >> 8) & 0xFF);
-    code[2] = (char) ((format >> 16) & 0xFF);
-    code[3] = (char) ((format >> 24) & 0xFF);
-    code[4] = '\0';
+    std::span<char> buffer;
+    CString code = CString::newUninitialized(4, buffer);
+    buffer[0] = static_cast<char>((format >> 0) & 0xFF);
+    buffer[1] = static_cast<char>((format >> 8) & 0xFF);
+    buffer[2] = static_cast<char>((format >> 16) & 0xFF);
+    buffer[3] = static_cast<char>((format >> 24) & 0xFF);
```

Comment:
This feels like it could be:

```suggestion
    auto code = FourCC(format & 0xFFFFFFFF).string();
```

Reply:
It doesn't work, I see other names with this, I guess `format & 0xFFFFFFFF` is not what we should be passing to FourCC

Reply:
It's not critical, but with the CString approach we save an allocation, this returns a std::array and then I have to create a String from it to pass it to makeString, while makeString can handle CString directly